### ISSUE TITLE
KeyError: 'token'

### ISSUE THREAD

**Issue by dsdshcym** [parkouss/subscope#1]
Hi! I discovered this project several days ago. And this is a great project. 

But when I used it today, I ran into this error:

```
ERROR:subscope.core:Error while searching subtitles for OpenSubtitles
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/subscope/core.py", line 54, in _search
    subtitles = source.search(movie_path, langs)
  File "/usr/local/lib/python2.7/site-packages/subscope/sources/opensubtitles.py", line 195, in search
    token = result['token']
KeyError: 'token'
```

And I've run the tests in the tests folder, with following results:

```
$ python3 -m unittest test_sources_opensubtitles.py
E/Users/dsdshcym/Github/subscope/subscope/tests/test_sources_opensubtitles.py:107: DeprecationWarning: Please use assertEqual instead.
  self.assertEquals(subs, expected)
.....
======================================================================
ERROR: test_download (test_sources_opensubtitles.TestOpenSubtitles)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/mock.py", line 1157, in patched
    return func(*args, **keywargs)
  File "/Users/dsdshcym/Github/subscope/subscope/tests/test_sources_opensubtitles.py", line 117, in test_download
    self.source.download({'link': 'http://sublink'}, stream)
  File "/usr/local/lib/python3.5/site-packages/subscope/sources/opensubtitles.py", line 211, in download
    stream.write(gzip.GzipFile(fileobj=StringIO(response.content)).read())
TypeError: initial_value must be str or None, not bytes

----------------------------------------------------------------------
Ran 6 tests in 0.247s

FAILED (errors=1)
```

```
$ python3 -m unittest test_sources_thesubdb.py
E
======================================================================
ERROR: test_sources_thesubdb (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_sources_thesubdb
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/loader.py", line 153, in loadTestsFromName
    module = __import__(module_name)
  File "/Users/dsdshcym/Github/subscope/subscope/tests/test_sources_thesubdb.py", line 21, in <module>
    from StringIO import StringIO
ImportError: No module named 'StringIO'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
```

**Comment by parkouss** [comment]
Oh, you should run the tests using "python3 setup.py test" as stated in the README - there is a step that use py2to3 to adapt the source code (this is a setuptools feature). Maybe it would be better to write the code compatible without this, but this is another story. :)

Thanks for the bug report! I will look into that - or I will be happy with a PR if you want to tackle this. :)

**Comment by parkouss** [comment]
I can reproduce it easily.
So it is failing here: https://github.com/parkouss/subscope/blob/master/subscope/sources/opensubtitles.py#L192.

If I print that result variable, I see:

``` python
{'status': '301 Moved (http://api.opensubtitles.org/xml-rpc)'}
```

But this is strange because this is the exact url used, e.g:

https://github.com/parkouss/subscope/blob/master/subscope/sources/opensubtitles.py#L180

And the doc (http://trac.opensubtitles.org/projects/opensubtitles/wiki/XMLRPC) still recommend this url.

**Comment by dsdshcym** [comment]
Sorry for running tests incorrectly. It passed all of the 37 tests after I ran `python3 setup.py test`.

Yes, I think that's the problem, too.
And I can still use subscope to download a subtitle if there is one from TheSubDB. So maybe opensubtitles is changing its API these days?

I'll keep an eye on this. Thanks for your reply!

**Comment by parkouss** [comment]
Hey! I had some time to look again, and investigated some more because it was still failing. Finally fixed:
https://github.com/parkouss/subscope/commit/90cf1cca46f30ee97f86491c65a2d96da646d9d7
So I released 1.0.2 with the fix, https://pypi.python.org/pypi/subscope/1.0.2

FYI, it was a bug in subscope, because it used "http://api.opensubtitles.org//xml-rpc". But the server never complained about it, until you saw the error. :)

Thanks a lot for the bug report ! Too bad I looked at it in details so late;

Anyway, this should now be fixed. Feel free to reopen the bug if you have any issue.

**Comment by dsdshcym** [comment]
@parkouss Thanks for your work! :smile:
